### PR TITLE
Set ROS_VERSION for rosdep in plugins v1

### PR DESCRIPTION
**Pull request by artivis**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

This enables ROS conditional dependencies in `package.xml` such as e.g.
```xml
<build_depend condition="$ROS_VERSION == 1">roscpp</build_depend>
<build_depend condition="$ROS_VERSION == 2">rclcpp</build_depend>
```

-----

### CODE REVIEW

**Comment by artivis** [comment]
@kyrofa tests are passing :+1:

File: tests/unit/plugins/v1/test_colcon.py [hunk 1]
```diff
@@ -70,12 +71,13 @@ class props:
         self.pip_mock.return_value.list.return_value = {}
 
     def assert_rosdep_setup(
-        self, rosdistro, package_path, rosdep_path, ubuntu_distro, base
+        self, rosdistro, rosversion, package_path, rosdep_path, ubuntu_distro, base
```
Comment (cjp256) [inline tests/unit/plugins/v1/test_colcon.py:74]: nitpick: should probably be `ros_version` for consistency with the rest, even if `rosdistro` is off

**Review by cjp256 — DISMISSED** [review]
Looks good, just one optional nitpick

File: tests/unit/plugins/v1/test_colcon.py (hunk 1 above)
Comment (artivis) [reply tests/unit/plugins/v1/test_colcon.py]: Fixed in 2799f6f.

**Comment by artivis** [comment]
The failing test, `google:ubuntu-16.04-64:tests/spread/general/package-repositories:test_apt_ppa`, seems unrelated to this PR.

**Review by cjp256 — APPROVED** [review]
Thanks, LGTM - I restarted the tests, but it is unrelated.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**